src/common/libflux/message_proto.c

```diff
@@ -8,51 +8,70 @@
  * SPDX-License-Identifier: LGPL-3.0
 \************************************************************/
 
+/* message_proto.c - marshal RFC 3 PROTO frame */
+
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif
 #include <errno.h>
-#include <stdbool.h>
-#include <string.h>
-#include <arpa/inet.h>
-#include <assert.h>
-#include <inttypes.h>
-
-#include "message.h"
-#include "message_private.h"
+
+#include "ccan/pushpull/pushpull.h"
+
 #include "message_proto.h"
 
-static void proto_set_u32 (uint8_t *data, int index, uint32_t val)
+static size_t fixed_size;
```

Comment:
This static variable could use a comment IMO. I'm still a little confused on how this will work in the broker with multiple threads possibly calling these protocol functions at the same time. I have no doubt it does work, I'm just confused how. A comment to soothe amateurs like myself might go a long way.

Reply:
Good point - this code would have to assume the `fixed_size` never changes.  It happens not to but that's not a great assumption given that the function takes a variable size buffer.  Eek bad.  Will fix.